### ISSUE TITLE
SIMD support is always disabled (at least under Windows)

### ISSUE THREAD

**Issue by db4** [tesseract-ocr/tesseract#2544]
### Environment

* **Tesseract Version**: 4.1
* **Commit Number**: db823d6589be3fd861291e75b343c5eaa0d9461e
* **Platform**: Windows 32-bit

### Current Behavior:

Commit 8a6fa452dc6a34e4fded2e7eee8dcf7951a8f403 has actually removed SIMD support on Windows:

https://github.com/tesseract-ocr/tesseract/blob/74145f0686656ecd0380cc7ccd648f80bbe0848e/src/arch/simddetect.cpp#L27-L29

Where `SSE4_1` and friends are expected to be defined? @stweil can you explain?

**Comment by zdenop** [comment]
How are you building tesseract? This variables are defined during configuration.

**Comment by stweil** [comment]
@db4, please run `tesseract --version`. If it reports SSE, AVX or AVX2, you have a binary and a CPU with SIMD support.

**Comment by db4** [comment]
> How are you building tesseract? This variables are defined during configuration.

Via [conan.io](https://conan.io) package https://github.com/bincrafters/conan-tesseract which I'm trying to adapt for branch 4.1.
Can you point to the exact place in CMakeLists.txt where these vars are defined?

> please run tesseract --version. If it reports SSE, AVX or AVX2, you have a binary and a CPU with SIMD support.

```
tesseract 4.1.0-rc4-26-gb101d
 leptonica-1.76.0 (Jul  2 2019, 14:06:15) [MSC v.1916 LIB Debug x86]
  libgif 5.1.4 : libjpeg 9c : libpng 1.6.34 : libtiff 4.0.9 : zlib 1.2.11
```

**Comment by db4** [comment]
Previous version on the same hardware:
```
tesseract 4.0.0
 leptonica-1.76.0 (Jul  2 2019, 14:00:36) [MSC v.1916 LIB Debug x64]
  libgif 5.1.4 : libjpeg 9c : libpng 1.6.34 : libtiff 4.0.9 : zlib 1.2.11
 Found AVX
 Found SSE
```

**Comment by zdenop** [comment]
We do not support conan build system - only cmake, autotools and sw (Windows).
configuration tools are responsible to checking availability of cpu features (see e.g. [cmake implementation](https://github.com/tesseract-ocr/tesseract/blob/74145f0686656ecd0380cc7ccd648f80bbe0848e/CMakeLists.txt#L195))

**Comment by db4** [comment]
> We do not support conan build system - only cmake

I do not expect you to support conan. But it is not needed: conan just runs CMake while building the package.
OK, I'll investigate the problem and if I find a problem on the CMake side I'll post additional details here.

**Comment by stweil** [comment]
@db4, maybe the Microsoft compiler only defines the `__AVX__` macro, but not `AVX` (and similar for the other two SIMD related macros `AVX2` and `SSE4_1`). Does it work for you when the conditionals in simddetect.cpp use the macro variants with underscores?

**Comment by db4** [comment]
I've found a particular problem. I'm running Windows in a virtual machine. Some time ago I changed Virtual CPU family/model but Windows 10 did not update its identifier in the registry accordingly (that you used to detect CPU ID). I've now synced it manually and now AVX/SSE 4.1 are successfully detected by CMake. (BTW, I think using the registry to detect CPU features is not a good CMake way. OpenCV does a similar job far more elegantly I think).

But there is still a problem I believe. `simddetect.cpp` detected AVX/SSE during runtime so one could use tesseract 4.0.0 binary compiled with AVX even on old hardware that did not support it. But now it's not possible. This
https://github.com/tesseract-ocr/tesseract/blob/b101d58621e511652df9bfdfe8b142450089dad5/cmake/OptimizeForArchitecture.cmake#L476-L482
and similar code forces use of AVX instructions everywhere. Tesseract executable simply won't start on a hardware without it (will crash with "invalid instruction" error)
